src/excmds.ts

```diff
@@ -3527,25 +3528,29 @@ export function firefoxsyncpush() {
-
- For example: `autocmd BeforeRequest https://www.bbc.co.uk/* () => ({redirectUrl: "https://old.reddit.com"})`. Note the brackets which ensure JavaScript returns a blocking response object rather than interpreting it as a block statement.
-
-*/
+ *
+ * @param event Curently, 'TriStart', 'DocStart', 'DocLoad', 'DocEnd', 'TabEnter', 'TabLeft', 'FullscreenChange', 'FullscreenEnter', 'FullscreenLeft', 'AuthRequired', 'BeforeRedirect', 'BeforeRequest', 'BeforeSendHeaders', 'Completed', 'ErrorOccured', 'HeadersReceived', 'ResponseStarted', and 'SendHeaders' are supported
+ *
+ * @param url For DocStart, DocEnd, TabEnter, and TabLeft: a JavaScript regex (e.g. `www\\.amazon\\.co.*`)
+ *
+ * We just use [URL.search](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search).
+ *
+ * For TriStart: A regular expression that matches the hostname of the computer
+ * the autocmd should be run on. This requires the native messenger to be
+ * installed, except for the ".*" regular expression which will always be
+ * triggered, even without the native messenger.
+ *
+ * For AuthRequired, BeforeRedirect, BeforeRequest, BeforeSendHeaders, Completed, ErrorOccured, HeadersReceived, ResponseStarted and SendHeaders, a [URL match pattern](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Match_patterns)
+ *
+ * @param excmd The excmd to run (use [[composite]] to run multiple commands), __except__ for AuthRequired, BeforeRedirect, BeforeRequest, BeforeSendHeaders, Completed, ErrorOccured, HeadersReceived, ResponseStarted and SendHeaders, events where it must be an inline JavaScript function which maps [details objects specific to the event](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest#Events) to [blocking responses](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/BlockingResponse). This JavaScript function will run in the background context.
+ *
+ * For example: `autocmd BeforeRequest https://www.bbc.co.uk/* () => ({redirectUrl: "https://old.reddit.com"})`. Note the brackets which ensure JavaScript returns a blocking response object rather than interpreting it as a block statement.
+ *
+ * For DocStart, DocLoad, DocEnd, TabEnter, TabLeft, AuthRequired, BeforeRedirect, BeforeRequest, BeforeSendHeaders, HeadersReceived, ResponceStarted and SendHeaders: a magic variable called `TRI_FIRED_MOZ_TABID` is also available which will be replaced with the Mozilla `tabId` of the window/tab associated with the fired event.
```

Comment:
So I have listed the events I believe are supported but I am not sure, I am going to need some help with that.

Reply:
This isn't quite right. It's any event that calls loadaucmds.

```
~/p/tridactyl (master|✔) $ rg -i loadaucmds
src/excmds.ts
1844:loadaucmds("DocStart")
1845:window.addEventListener("pagehide", () => loadaucmds("DocEnd"))
1846:window.addEventListener("DOMContentLoaded", () => loadaucmds("DocLoad"))
1849:    loadaucmds("FullscreenChange")
1851:        loadaucmds("FullscreenEnter")
1853:        loadaucmds("FullscreenLeft")
1870:export async function loadaucmds(cmdType: "DocStart" | "DocLoad" | "DocEnd" | "TabEnter" | "TabLeft" | "FullscreenEnter" | "FullscreenLeft" | "FullscreenChange") {

src/background.ts
157:            .messageTab(curTab, "excmd_content", "loadaucmds", ["TabLeft"])
162:        .messageTab(curTab, "excmd_content", "loadaucmds", ["TabEnter"])
```

So that's the Doc* events, the Tab* events and Fullscreen* events.

Reply:
I didn't realize that some events did not call loadautocmds, its fixed now :D.